src/middlewares/authMiddleware.js

```diff
@@ -44,3 +41,22 @@ export async function signInMiddleware(req, res) {
     res.sendStatus(500);
   }
 }
+
+export async function authValidation(req, res, next) {
+  const { authorization } = req.headers;
+  if (!authorization) return res.sendStatus(401);
+  const token = authorization.replace("Bearer ", "");
+
+  try {
+    const session = await db.collection("sessions").findOne({ token });
+    if (!session) return res.sendStatus(401);
+    const user = await db.collection("users").findOne({ _id: session.userId });
+    if (!user) return res.status(404).send("User does not exist");
+
+    res.locals.session = session;
+
+    next();
```

Comment:
O next precisa ser fora do escopo do trycatch

```suggestion
try {
} catch (err) {
}
    next();
```
O next ser chamado dentro do escopo do try catch atrapalha o fluxo de identificação de erro.